### ISSUE TITLE
Import: check if UID already exists, check which one is newest and update if necessary.

### ISSUE THREAD

**Issue by georgehrke** [nextcloud/calendar#1508]

**Comment by tcitworld** [comment]
I'm guessing you mean when importing data in an existing calendar, it makes little sense when creating a new one (UIDs are only unique in their calendar - even though there should be globally unique).

**Comment by georgehrke** [comment]
Yes, only for existing calendars.
Although it would be nice to check for the UID globally, but that's something for way later.

**Comment by r2evans** [comment]
(Is this why I've been beating my head against a wall trying to update events via import?)

The iCal spec ([ref 5545](https://tools.ietf.org/html/rfc5545)) says that event updates should occur based on UID and SEQUENCE (and/or based on DTSTAMP and/or LAST-MODIFIED). If we're doing an import, shouldn't one or more of those fields be referenced?

It's a common enough use-case (I believe) for a user to upload an updated (externally-provided) `.ics` file and expect that updated VEVENTs are correct in the calendar app.

(I think my only workaround at the moment is to either delete all updated events and reimport them, ignoring all import failures. Sadly, this means I need to know which have been changed. Lacking that, I often end up deleting all events in that category and time-span and re-importing everything. It works well enough and all other users of that calendar eventually get what they need, I think ... but it defies the RFC/spec, and I'm not 100% certain that some client won't out-smart this somehow, such as based on UID.)

**Comment by Whisprin** [comment]
Currently the API returns this error when a user tries to import an updated event invitation.
```
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\BadRequest</s:exception>
  <s:message>Calendar object with uid already exists in this calendar collection.</s:message>
</d:error>
```

In the calendar web interface it looks like this
![image](https://github.com/nextcloud/calendar/assets/3153136/2f3f8d5b-fcef-4492-8b38-08642154927f)
There's a [separate issue](https://github.com/nextcloud/calendar/issues/2065) to improve those error messages.

**Comment by r2evans** [comment]
A related issue in server https://github.com/nextcloud/server/issues/30096 was closed last year, though it is still happening. Can a dev comment please?

**Comment by beedell-roke** [comment]
@r2evans, is [`help.nextcloud.com/t/233713/2`](https://help.nextcloud.com/t/how-should-i-ascertain-whether-an-http-400-that-is-reported-by-davx5-every-time-it-polls-a-specific-ics-file-is-its-fault-or-the-nextcloud-provider-s/233713/2?u=rokejulianlockhart) an example of it?

**Comment by SebastianKrupinski** [comment]
Hi All,

There is now a new calendar import command that can do more that the UI import does.

https://github.com/nextcloud/server/pull/51925

There should also be new import functionality in the calendar once this is merged.

https://github.com/nextcloud/server/pull/55178

**Comment by beedell-roke** [comment]
@SebastianKrupinski, those don't appear to be available for those without control over their instances (anyone who registered at what [`nextcloud.com/partners`](https://nextcloud.com/partners/) lists), though.

**Comment by SebastianKrupinski** [comment]
@RokeJulianLockhart 

The second PR will be

**Comment by r2evans** [comment]
> [@r2evans](https://github.com/r2evans), is [`help.nextcloud.com/t/233713/2`](https://help.nextcloud.com/t/how-should-i-ascertain-whether-an-http-400-that-is-reported-by-davx5-every-time-it-polls-a-specific-ics-file-is-its-fault-or-the-nextcloud-provider-s/233713/2?u=rokejulianlockhart) an example of it?

I'm not sure

**Comment by marcosmarquesnic** [comment]
THEEY SOLVED THE PROBLEMMMMMM THANKS NEXTCLODO
